bff/pkg/server/server.go

```diff
@@ -319,112 +331,117 @@ func metrics_handler(route_configs map[string]RouteConfig, dbIdentifiers []strin
-				return
-			}
+func activity_handler(metrics_service metrics.Service, validate *validator.Validate) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		params := ActivityParams{
+			DbIdentifier:      r.URL.Query().Get("dbidentifier"),
+			DatabaseList:      r.URL.Query().Get("database_list"),
+			Start:             r.URL.Query().Get("start"),
+			End:               r.URL.Query().Get("end"),
+			Step:              r.URL.Query().Get("step"),
+			Legend:            r.URL.Query().Get("legend"),
+			Dim:               r.URL.Query().Get("dim"),
+			FilterDim:         r.URL.Query().Get("filterdim"),
+			FilterDimSelected: r.URL.Query().Get("filterdimselected"),
+			Limit:             r.URL.Query().Get("limitdim"),
+			LimitLegend:       r.URL.Query().Get("limitlegend"),
 		}
 
-		promQLInput := PromQLInput{
-			DatabaseList:      database_list,
-			Start:             start,
-			End:               end,
-			Step:              step,
-			Legend:            legend,
-			Dim:               dim,
-			FilterDim:         filterdim,
-			FilterDimSelected: filterdimselected,
-			Limit:             limitDim,
-			LimitLegend:       limitLegend,
-			Offset:            "", // TODO not in query
+		if err := validate.Struct(params); err != nil {
+			fmt.Println("Validation failed: ", err)
+			http.Error(w, fmt.Sprintf("Validation failed: %v", err), http.StatusBadRequest)
+			return
 		}
 
-		fmt.Println("PromQLInput: ", promQLInput)
+		now := time.Now()
 
-		query, err := GenerateActivityCubePromQLQuery(promQLInput)
+		startTime, err := parseTimeParameter(params.Start, now)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
 
-		fmt.Println("query: ", query)
-		now := time.Now()
-
-		startTime, err := parseTimeParameter(start, now)
+		endTime, err := parseTimeParameter(params.End, now)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
 
-		var endTime time.Time
-		if end != "" {
-			endTime, err = parseTimeParameter(end, now)
+		limitValue := 0
+		limitLegendValue := 0
+		offsetValue := 0
+
+		if params.Limit != "" {
+			limitValue, err = strconv.Atoi(params.Limit)
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusBadRequest)
 				return
 			}
-		} else {
-			endTime = now
 		}
 
-		if startTime.After(endTime) {
-			http.Error(w, "Parameter 'end' must be greater than 'start'", http.StatusBadRequest)
+		if params.LimitLegend != "" {
+			limitLegendValue, err = strconv.Atoi(params.LimitLegend)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+		}
+
+		dbListEscaped := strconv.Quote(params.DatabaseList)[1 : len(params.DatabaseList)-1]
+
+		filterDimSelectedEscaped := ""
+
+		if params.FilterDimSelected != "" {
+			filterDimSelectedEscaped = strconv.Quote(params.FilterDimSelected)[1 : len(params.FilterDimSelected)-1]
+		}
```

Comment:
This is problematic, as you shouldn't use `len(params.DatabaseList)` and `len(params.FilterDimSelected)`. Please consider doing something like this:


```suggestion
	dbListEscaped := ""
	if params.DatabaseList != "" {
		// Use strconv.Quote and ensure proper escaping
		dbListEscaped = strconv.Quote(params.DatabaseList)
		// Remove the first and last quotes
		dbListEscaped = dbListEscaped[1 : len(dbListEscaped)-1]
	}

	filterDimSelectedEscaped := ""
	if params.FilterDimSelected != "" {
		// Use strconv.Quote and ensure proper escaping
		filterDimSelectedEscaped = strconv.Quote(params.FilterDimSelected)
		// Remove the first and last quotes
		filterDimSelectedEscaped = filterDimSelectedEscaped[1 : len(filterDimSelectedEscaped)-1]
	}
```

Reply:
addressed in c26e61582aeb1fc8bda0c60ee486c6d9c665f0d4